### PR TITLE
Github Actions: move PR checks from static tests to own workflow

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This moves PR-related checks from the static tests to the same workflow, as the `check-label` action (now renamed to `check-pr.yml`).

### Testing procedure
Checks should pass and also made required.

### Issues/PRs references
None, but #15563 would need to be adopted when merged.

### CODE REVIEW

**Comment by miri64** [comment]
:-/

**Comment by miri64** [comment]
OK, that action is too happy ^^" (it succeeds even though there are broken commit messages in here). Will check next week!

File: .github/workflows/check-pr.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - name: Run checks
+      env:    # enforce deactivation ubuntu label check
+        TRAVIS_PULL_REQUEST:
+        CI_PULL_NR:
+      run: |
+        ./dist/tools/pr_check/pr_check.sh "${{ github.base_ref }}"
+        RES1=$?
+        ./dist/tools/commit-msg/check.sh "${{ github.base_ref }}"
+        RES2=$?
+        [ "${RES1}" -eq 0 ] && [ "${RES2}" -eq 0 ]
```
Comment (aabadie) [inline .github/workflows/check-pr.yml:38]: You could use a matrix strategy with fail-fast set to false. If you rename pr_check.sh to check.sh, the items in the matrix would just be `["pr_check", "commit-msg"]`.

File: .github/workflows/check-pr.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        # in master (default behavior in GH actions)
+        # See https://github.com/actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+    - name: Fetch base branch
+      run:
+        git fetch origin '${{ github.base_ref }}:${{ github.base_ref }}'
+    - name: Run checks
+      env:    # enforce deactivation ubuntu label check
+        TRAVIS_PULL_REQUEST:
```
Comment (aabadie) [inline .github/workflows/check-pr.yml:31]: This variable could be renamed to better reflect that this is running on GH actions, or maybe rename it to something more generic (`CI_PULL_REQUEST` ?)

**Review by aabadie — COMMENTED** [review]
A few comments :)

File: .github/workflows/check-pr.yml (hunk 1 above)
Comment (miri64) [reply .github/workflows/check-pr.yml]: Yep just was doing that :-).

File: .github/workflows/check-pr.yml (hunk 2 above)
Comment (miri64) [reply .github/workflows/check-pr.yml:35]: The variable is defined within `check_pr`. This `env` is mostly to ensure that it isn't set by some spurious environment ;-)'

Comment (aabadie) [reply .github/workflows/check-pr.yml:35]: Yes I understand that. pr_check.sh can be updated of course.

Comment (miri64) [reply .github/workflows/check-pr.yml:35]: There is some clean-up required in that script in any case. Should we put that to a follow-up maybe and stick to the current workaround for now? I updated the comment to be a bit clearer on that.

**Comment by miri64** [comment]
Rebased and should be ready for review now.

**Comment by miri64** [comment]
Now that the static tests are fixed in master, those checks really should go to their own action, as especially the commit message checkers yields some non-sensical noise when ran on the branch (non-sensical since it doesn't make sense to fix commit messages in upstream ;-)).

**Comment by aabadie** [comment]
Hmm, the static tests are successful even though some or failing.

**Comment by miri64** [comment]
> Hmm, the static tests are successful even though some or failing.

Huh? Which one should be failing? Or do you mean I should add a test commit that induces an static test error?

File: .github/workflows/check-pr.yml (hunk 2 above)
Comment (miri64) [reply .github/workflows/check-pr.yml]: Done that cleanup.

**Comment by aabadie** [comment]
> Which one should be failing? Or do you mean I should add a test commit that induces an static test error?

This one: https://github.com/RIOT-OS/RIOT/pull/15564/checks?check_run_id=1664303991
This PR introduces a trailing whitespace and an empty new line at EOF. The corresponding static tests are failing but the global status is ok.

**Comment by miri64** [comment]
The job succeeded, since `whitespacecheck` now pipes its results to `${LOG}`, resulting in the error not being recognized in `$?`. I added a fix, to `github-annotate.sh`, but that is a bashism. Still searching for a POSIX shell compliant fix.

**Comment by miri64** [comment]
Mhh yeah... [this did not have the desired effect...](https://github.com/RIOT-OS/RIOT/pull/15564/checks?check_run_id=1668246959#step:5:14):

```
 	./dist/tools/whitespacecheck/check.sh: 8: set: Illegal option -o pipefail
```

**Comment by miri64** [comment]
I think this is the cleaner approach. Still a bashism, but not enforcing bashisms on all users of `github_annotate.sh`. The only non-bashism [I found](https://unix.stackexchange.com/a/70675) led to a pretty unreadable script.

**Comment by miri64** [comment]
Since this is an unrelated failure already broken in master, I can pull the piggy-backed fix out of this PR. But we also can keep it here. You decide.

**Comment by aabadie** [comment]
I think it's ok to keep it here. I'm also perfectly fine with using bash. Just squash and remove the "REMOVE ME" commits, and then we are good.

**Comment by miri64** [comment]
> I think it's ok to keep it here. I'm also perfectly fine with using bash. Just squash and remove the "REMOVE ME" commits, and then we are good.

Done and rebased.

**Review by aabadie — APPROVED** [review]
All good for me. ACK

We just have to make this new check required.

**Comment by miri64** [comment]
> We just have to make this new check required.

Done

**Comment by aabadie** [comment]
Hmm, something is wrong with check-commits (pull_request_review)... it seems it checks on the whole git history.

**Comment by miri64** [comment]
grmpf...

**Comment by aabadie** [comment]
Can we have the check-commits check only performed on pull_request events ?

**Review by aabadie — APPROVED** [review]
ACK2

**Comment by aabadie** [comment]
Ok, so check-commits is ignored on review but still triggered on regular pull_request events.

**Comment by miri64** [comment]
Made the actions required now :-).

**Pull request merged**